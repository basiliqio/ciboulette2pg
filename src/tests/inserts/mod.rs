use super::*;

mod inserts_success;
