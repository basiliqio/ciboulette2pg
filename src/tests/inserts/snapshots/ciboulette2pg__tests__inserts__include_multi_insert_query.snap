---
source: src/tests/inserts/mod.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
WITH "cte_peoples_insert_0" AS (
  INSERT INTO
    "public"."peoples" (
      "age",
      "first-name",
      "gender",
      "last-name",
      "twitter"
    )
  VALUES
    ($1, $2, $3, $4, $5) RETURNING *
),
"cte_peoples_data_1" AS (
  SELECT
    "cte_peoples_insert_0"."id" :: TEXT AS "id",
    "cte_peoples_insert_0"."id" AS "main_id",
    NULL :: TEXT AS "related_id",
    NULL :: TEXT AS "related_type",
    $6 :: TEXT AS "type",
    JSON_BUILD_OBJECT(
      $7,
      "cte_peoples_insert_0"."age",
      $8,
      "cte_peoples_insert_0"."first-name",
      $9,
      "cte_peoples_insert_0"."gender",
      $10,
      "cte_peoples_insert_0"."last-name",
      $11,
      "cte_peoples_insert_0"."twitter"
    ) AS "data",
    "cte_peoples_insert_0"."favorite_color" AS "rel_favorite_color"
  FROM
    "cte_peoples_insert_0"
),
"cte_rel_articles_data_2" AS (
  SELECT
    "public"."articles"."id" :: TEXT AS "id",
    "public"."articles"."id" AS "main_id",
    "cte_peoples_data_1"."id" :: TEXT AS "related_id",
    $12 :: TEXT AS "related_type",
    $13 :: TEXT AS "type",
    JSON_BUILD_OBJECT(
      $14,
      "public"."articles"."body",
      $15,
      "public"."articles"."title"
    ) AS "data"
  FROM
    "public"."articles"
    INNER JOIN "public"."people-article" ON "public"."people-article"."article_id" = "public"."articles"."id"
    INNER JOIN "cte_peoples_data_1" ON "cte_peoples_data_1"."main_id" = "public"."people-article"."people_id"
) (
  SELECT
    "cte_peoples_data_1"."id",
    "cte_peoples_data_1"."type",
    "cte_peoples_data_1"."data",
    "cte_peoples_data_1"."related_type",
    "cte_peoples_data_1"."related_id"
  FROM
    "cte_peoples_data_1"
)
UNION
ALL (
  SELECT
    "cte_rel_articles_data_2"."id",
    "cte_rel_articles_data_2"."type",
    "cte_rel_articles_data_2"."data",
    "cte_rel_articles_data_2"."related_type",
    "cte_rel_articles_data_2"."related_id"
  FROM
    "cte_rel_articles_data_2"
);
